src/go/supply/supply.go

```diff
@@ -93,9 +95,13 @@ func (gs *Supplier) SelectVendorTool() error {
 		return errors.New(".godir deprecated")
 	}
 
-	if exists, err := libbuildpack.FileExists(filepath.Join(gs.Stager.BuildDir(), "go.mod")); err != nil {
+	isGoMod, err := libbuildpack.FileExists(filepath.Join(gs.Stager.BuildDir(), "go.mod"))
+	if err != nil {
 		return err
-	} else if exists {
+	}
+	if isGoMod {
+		gs.Log.BeginStep("Checking go.mod file")
+		gs.GoMod.Load(filepath.Join(gs.Stager.BuildDir(), "go.mod"))
```

Comment:
And here should be done the handling of any error returned by `Load`.

Reply:
Yep, makes sense. We should warn and not select the version.